plutus-core/plutus-core/test/Pretty/Golden/Readable/StdLib/Data/Data/caseData.golden

```diff
@@ -5,20 +5,12 @@
      (fList : list data -> r)
      (fI : integer -> r)
      (fB : bytestring -> r) ->
-      chooseData
+      caseData
         {unit -> r}
         d
-        (\(u : unit) ->
-           (/\a b c ->
-              \(f : a -> b -> c) (p : pair a b) ->
-                f (fstPair {a} {b} p) (sndPair {a} {b} p))
-             {integer}
-             {list data}
-             {r}
-             fConstr
-             (unConstrData d))
-        (\(u : unit) -> fMap (unMapData d))
-        (\(u : unit) -> fList (unListData d))
-        (\(u : unit) -> fI (unIData d))
-        (\(u : unit) -> fB (unBData d))
+        (\(i : integer) (ds : list data) (u : unit) -> fConstr i ds)
+        (\(es : list (pair data data)) (u : unit) -> fMap es)
+        (\(ds : list data) (u : unit) -> fList ds)
+        (\(i : integer) (u : unit) -> fI i)
+        (\(b : bytestring) (u : unit) -> fB b)
```

Comment:
I don't understand why the `unit`s are needed if computation is already under a function type.

Reply:
Right, I think I just mechanically updated the previous definition which was in terms of `chooseData` where `unit`s were needed without realizing that they're no longer needed. Thank you, will fix.

Reply:
In fact, we don't need anything more complicated than

```
-- | Pattern matching over 'Data' inside PLC.
--
-- > \(d : data) -> /\(r :: *) -> caseData {r} d
caseData :: TermLike term TyName Name DefaultUni DefaultFun => term ()
caseData = runQuote $ do
    r <- freshTyName "r"
    d <- freshName "d"
    return
        . lamAbs () d dataTy
        . tyAbs () r (Type ())
        . apply () (tyInst () (builtin () CaseData) $ TyVar () r)
        $ var () d
```